sdk/eventhub/azure-eventhubs/azure/eventhub/common.py

```diff
@@ -152,27 +153,50 @@ def _trace_link_message(self, parent_span=None):
                 if traceparent:
                     current_span.link(traceparent)
 
+    def _get_runtime_info(self):
+        if self._runtime_info:
+            return self._runtime_info
+
+        if self.message.delivery_annotations:
+            self._runtime_info = {
+                "sequence_number":
+                    self.message.delivery_annotations.get(EventData.PROP_LAST_ENQUEUED_SEQUENCE_NUMBER, None),
+                "offset":
+                    self.message.delivery_annotations.get(EventData.PROP_LAST_ENQUEUED_OFFSET, None),
+                "enqueued_time":
+                    self.message.delivery_annotations.get(EventData.PROP_LAST_ENQUEUED_TIME_UTC, None),
+                "retrieval_time":
+                    self.message.delivery_annotations.get(EventData.PROP_RUNTIME_INFO_RETRIEVAL_TIME_UTC, None)
+            }
+            return self._runtime_info
+
+        return None
+
     @staticmethod
     def _from_message(message):
         # pylint:disable=protected-access
         event_data = EventData(body='')
         event_data.message = message
-        event_data._msg_properties = message.properties
-        event_data._annotations = message.annotations
-        event_data._app_properties = message.application_properties
-        event_data._delivery_annotations = message.delivery_annotations
-        if event_data._delivery_annotations:
-            event_data._runtime_info = {
+        event_data._need_further_parse = True
+        return event_data
+
+    def _parse_message_properties(self):
+        self._msg_properties = self.message.properties
+        self._annotations = self.message.annotations
+        self._app_properties = self.message.application_properties
+        self._delivery_annotations = self.message.delivery_annotations
+        if self._delivery_annotations:
+            self._runtime_info = {
```

Comment:
method _get_runtime_info() checkes is self-sufficient to populate runtime info when it's None and return it when it's not None. No need to populate it here?

Reply:
you're right, just removed the code

Reply:
you're right, just removed the code